tracer/src/Datadog.Trace/Span.cs

```diff
@@ -115,7 +115,7 @@ internal ulong RootSpanId
 
         internal bool IsRootSpan => Context.TraceContext?.RootSpan == this;
 
-        internal bool IsTopLevel => Context.Parent == null || Context.Parent.ServiceName != ServiceName;
+        internal bool IsTopLevel => Context.Parent == null || Context.Parent.SpanId == 0 || Context.Parent.ServiceName != ServiceName;
```

Comment:
NOTE: I found this bug during testing. If a span's parent has `SpanId: 0`, then the span doesn't have a parent and it is a service-entry (aka top-level) span.

We should also fix `IsRootSpan`, but I'll look into that in a separate PR.